### PR TITLE
Fix Test: System.Data.Tests.SqlTypes.SqlStringTest/SqlDateTimeToSqlString

### PR DESCRIPTION
**Pull request by nchikanov**
Implement fixes based on #23282

### CODE REVIEW

File: src/System.Data.Common/tests/System/Data/DataSetTest.cs [hunk 1]
```diff
@@ -37,25 +37,19 @@
...
 using System.Globalization;
 using System.Text;
+using System.Diagnostics;
 
 namespace System.Data.Tests
 {
-    public class DataSetTest : DataSetAssertion, IDisposable
+    public class DataSetTest : RemoteExecutorTestBase
     {
         private CultureInfo _currentCultureBackup;
```
Comment (danmosemsft) [inline src/System.Data.Common/tests/System/Data/DataSetTest.cs]: this is dead

File: src/System.Data.Common/tests/System/Data/DataSetTest.cs [hunk 2]
```diff
@@ -1565,16 +1567,21 @@ public void WriteXmlModeSchema()
         [Fact]
         public void WriteXmlModeSchema1()
         {
-            string SerializedDataTable =
+            RemoteInvoke(() =>
+            {
+                CultureInfo.CurrentCulture = new CultureInfo("fi-FI");
+
```
Comment (danmosemsft) [inline src/System.Data.Common/tests/System/Data/DataSetTest.cs]: extraneous line

File: src/System.Data.Common/tests/System/Data/DataTableReadXmlSchemaTest.cs [hunk 3]
```diff
@@ -50,17 +50,6 @@ private DataSet CreateTestSet()
 
         private CultureInfo _currentCultureBackup;
```
Comment (danmosemsft) [inline src/System.Data.Common/tests/System/Data/DataTableReadXmlSchemaTest.cs]: dead

**Comment by danmosemsft** [comment]
fixed build break

**Comment by nchikanov** [comment]
test Linux x64 Release Build 
test OSX x64 Debug Build 
test UWP CoreCLR x64 Debug Build

**Comment by danmosemsft** [comment]
@dotnet-bot test UWP CoreCLR x64 Debug Build  (fixed test now)

**Comment by danmosemsft** [comment]
OSX failure:
```
[osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest] Running shell script
+ HOME=/Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/tempHome
+ ./init-tools.sh
Installing dotnet cli...
curl: (7) Failed to connect to dotnetcli.azureedge.net port 443: Operation timed out
tar: Error opening archive: Failed to open '/Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/Tools/dotnetcli/dotnet.tar'
Restoring BuildTools version 2.0.0-prerelease-01914-02...
./init-tools.sh: line 101: /Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/Tools/dotnetcli/dotnet: No such file or directory
ERROR: Could not restore build tools correctly.
Please check the detailed log that follows.
Running: /Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/init-tools.sh
Installing 'https://dotnetcli.azureedge.net/dotnet/Sdk/2.0.0-preview1-005977/dotnet-dev-osx-x64.2.0.0-preview1-005977.tar.gz' to '/Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/Tools/dotnetcli/dotnet.tar'
Running: /Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/Tools/dotnetcli/dotnet restore "/Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/init-tools.msbuild" --no-cache --packages /Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/packages --source https://dotnet.myget.org/F/dotnet-buildtools/api/v3/index.json /p:BuildToolsPackageVersion=2.0.0-prerelease-01914-02
Initializing BuildTools...
chmod: /Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/packages/microsoft.dotnet.buildtools/2.0.0-prerelease-01914-02/lib/init-tools.sh: No such file or directory
./init-tools.sh: line 113: /Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/packages/microsoft.dotnet.buildtools/2.0.0-prerelease-01914-02/lib/init-tools.sh: No such file or directory
ERROR: An error occurred when trying to initialize the tools.
Please check the detailed log that follows.
Running: /Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/init-tools.sh
Installing 'https://dotnetcli.azureedge.net/dotnet/Sdk/2.0.0-preview1-005977/dotnet-dev-osx-x64.2.0.0-preview1-005977.tar.gz' to '/Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/Tools/dotnetcli/dotnet.tar'
Running: /Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/Tools/dotnetcli/dotnet restore "/Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/init-tools.msbuild" --no-cache --packages /Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/packages --source https://dotnet.myget.org/F/dotnet-buildtools/api/v3/index.json /p:BuildToolsPackageVersion=2.0.0-prerelease-01914-02
Running: /Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/packages/microsoft.dotnet.buildtools/2.0.0-prerelease-01914-02/lib/init-tools.sh /Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest /Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/Tools/dotnetcli/dotnet /Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/Tools
[Pipeline] }
```

@dotnet/dnceng can you plase look?

**Comment by ravimeda** [comment]
test OSX x64 Debug Build

**Pull request merged**